ZipArchive.cs

```diff
@@ -248,20 +376,115 @@ public ZipEntry AddFile (string sourcePath, string archivePath = null, EntryPerm
+		/// <param name="data">A stream containing the data to add to the archive</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, Stream data,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddStream (data, entryName, method: method);
+		}
+
+		/// <summary>
+		/// Adds the byte[] data to the archive
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="data">The draw byte[] array to add to the entry</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, byte [] data,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddEntry (data, entryName, method: method);
+		}
+
+		/// <summary>
+		/// Adds the text provided as a new entry within the archive. This is useful 
+		/// for when you want to just add plain text to the archive. For example a
+		/// README/LICENCE file which needs to be pre-processed rather than just adding
+		/// the raw file.
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="text">The text to add to the entry</param>
+		/// <param name="encoding">The Encoding to use for the data.</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, string text, Encoding encoding,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddEntry (entryName, encoding.GetBytes (text), method);
+		}
+
+		/// <summary>
+		/// Add a list of files to the archive
+		/// </summary>
+		/// <param name="fileNames">An IEnumerable<stirng> of files to add</param>
+		/// <param name="directoryPathInZip">The root directory path in archive.</param>
+		public void AddFiles (IEnumerable<string> fileNames, string directoryPathInZip)
+		{
+			foreach (var file in fileNames) {
+				AddFile (file, archivePath: directoryPathInZip);
+			}
+		}
+
+		/// <summary>
+		/// Checks to see if an entryName exists in the archive. This is done case-insensitively
+		/// </summary>
+		/// <returns>Returns true if the entry exists, false otherwise</returns>
+		/// <param name="entryName">The name of the entry to check for</param>
+		public bool ContainsEntry (string entryName)
+		{
+			return this.Any (x => string.Compare (x.Name, entryName, StringComparison.OrdinalIgnoreCase) == 0);
+		}
+
+		/// <summary>
+		/// Recursively Add an entire directory structure to the archive
+		/// </summary>
+		/// <param name="folder">The root of the directory to add</param>
+		/// <param name="folderInArchive">The root name of the folder in the zip.</param>
+		/// <param name="method">The compresison method to use when adding files</param>
+		public void AddDirectory (string folder, string folderInArchive, CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			string root = folderInArchive;
```

Comment:
Add null checks. Passing a `null` to `Directory.GetFiles` throws an exception. Also, `folderInArchive` should be required to be non-null